### ISSUE TITLE
Error "Incorrect permissions on pwfile" on using ssh kitten

### ISSUE THREAD

**Issue by soumitradev** [kovidgoyal/kitty#6455]
Whenever I use `ssh` normally, it works just fine, except that I run into the issue where the server sometimes errors with `'xterm-kitty': unknown terminal type`. To use ssh on kitty properly, I understand that I need to use the ssh kitten. However, when I use the ssh kitten, I get an error: `Incorrect permissions on pwfile`.

Verbose log for ssh:
```
❯ kitty +kitten ssh -vvv server                                                                                                                           (base) 
OpenSSH_9.0p1 Ubuntu-1ubuntu8.2, OpenSSL 3.0.8 7 Feb 2023
debug1: Reading configuration data /home/soumitradev/.ssh/config
debug1: /home/soumitradev/.ssh/config line 7: Applying options for server
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug2: resolve_canonicalize: hostname 125.22.xxx.xxx is address
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts' -> '/home/soumitradev/.ssh/known_hosts'
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts2' -> '/home/soumitradev/.ssh/known_hosts2'
debug1: auto-mux: Trying existing master
debug2: fd 3 setting O_NONBLOCK
debug2: mux_client_hello_exchange: master version 4
debug3: mux_client_forwards: request forwardings: 0 local, 0 remote
debug3: mux_client_request_session: entering
debug3: mux_client_request_alive: entering
debug3: mux_client_request_alive: done pid = 33923
debug3: mux_client_request_session: session request sent
debug1: mux_client_request_session: master session id: 2
Incorrect permissions on pwfile
debug3: mux_client_read_packet: read header failed: Broken pipe
debug2: Received exit status from master 1
Shared connection to 125.22.xxx.xxx closed.
```

```
kitty 0.26.5 created by Kovid Goyal
Linux soumitradev 6.2.0-25-generic #25-Ubuntu SMP PREEMPT_DYNAMIC Fri Jun 16 17:05:07 UTC 2023 x86_64
Ubuntu 23.04 soumitradev /dev/tty

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=23.04
DISTRIB_CODENAME=lunar
DISTRIB_DESCRIPTION="Ubuntu 23.04"
Running under: X11
Frozen: True
Paths:
  kitty: /usr/bin/kitty.app/bin/kitty
  base dir: /usr/bin/kitty.app/lib/kitty
  extensions dir: /usr/bin/kitty.app/lib/kitty-extensions
  system shell: /usr/bin/fish
Loaded config files:
  /etc/xdg/kitty/kitty.conf
  /home/soumitradev/.config/kitty/kitty.conf

Config options different from defaults:
active_tab_font_style   (True, False)
copy_on_select          clipboard
cursor_shape            2
font_family             RobotoMono Nerd Font
font_size               14.0
scrollback_lines        10000
tab_bar_edge            1
tab_bar_style           slant
```

**Additional context**
I can't seem to reproduce the issue with `kitty --config NONE`.

**Comment by kovidgoyal** [comment]
That will be because you have changed something on your system, possibly
the umask as a result of which files in shared memory have incorrect
permissions. Either fix that, or use one of the alternate ways to copy
the terminfo to your server, as described in the docs.

**Comment by soumitradev** [comment]
Is there a way I can figure out which file exactly has the incorrect umask? @kovidgoyal

**Comment by soumitradev** [comment]
@kovidgoyal Is there a way to figure out which file exactly has incorrect permissions on it?

**Comment by kovidgoyal** [comment]
It's not a file. kitty creates a  shared memory file to pass data to
the kitten. that file is being created with permissions other that owner
only read/write which indicates the umask on the parent dir or the
global umask is off

**Comment by soumitradev** [comment]
@kovidgoyal My global umask seems fine, it's `0002`, but which directory do you mean when you say "parent directory"? My `.ssh` directory, and all files inside of it have all the correct perms.

**Comment by kovidgoyal** [comment]
/dev/shm

**Comment by kovidgoyal** [comment]
kitty creates a file under /dev/shm/ with permissions 0o600 and later when reading from that file checks that the permissions are stil 0o600. Something on your system is causing that to not be true.

**Comment by soumitradev** [comment]
@kovidgoyal Running `ls -la` on `/dev/shm` in a loop, and trying the ssh kitten in another windows gives me this:
![image](https://github.com/kovidgoyal/kitty/assets/36817910/4a8b5370-5003-407c-85fe-f3c87acb8164)

**Comment by kovidgoyal** [comment]
That does not tell us anything, whatever is changing the permissions could easily be operating between ls invocations.

**Comment by kovidgoyal** [comment]
See this commit should tell us what the wrong permissions are https://github.com/kovidgoyal/kitty/commit/cf0d30808fe7a38279793b0d7f0568c1c0d4f87e

**Comment by soumitradev** [comment]
I'll try this, thanks

**Comment by soumitradev** [comment]
@kovidgoyal Hi, this is now resolved. It was quite stupid, but having to install the nightly binary worked for me. I realized when trying to run the nightly binary, that kitty has been running out of `/usr/bin/kitty.app/` this entire time, while I thought it was running out of `~/.local/kitty.app`. I'm not sure what caused this issue, but moment I made sure kitty was running out of `~/.local/kitty.app`, everything instantly solved itself. My guess is that something was broken in version 0.26.5, but is fixed in 0.29.0. Quite surprised to realize that my kitty hadn't been updating this entire time.

Thanks for the prompt response. I really enjoy this piece of software, and appreciate what you're doing.
